lib/xwax/timecoder.c

```diff
@@ -435,7 +437,19 @@ void timecoder_init(struct timecoder *tc, struct timecode_def *def,
     tc->forwards = 1;
     init_channel(tc->def, &tc->primary);
     init_channel(tc->def, &tc->secondary);
-    pitch_init(&tc->pitch, tc->dt);
+
+    tc->use_legacy_pitch_filter = pitch_estimator; /* Switch for pitch filter type */
+
+    if (tc->use_legacy_pitch_filter) {
+        pitch_init(&tc->pitch, tc->dt);
+    } else {
+        pitch_kalman_init(&tc->pitch_kalman, tc->dt,
+                KALMAN_COEFFS(1e-8, 10.0), /* stable mode */
+                KALMAN_COEFFS(1e-4, 1e-1), /* medium mode */
+                KALMAN_COEFFS(1e-1, 1e-4), /* reactive mode */
+                6e-4,   /* medium threshold  */
```

Comment:
The threshold is compared with the Inovation, the diffrence of the predicted x with the new x.  
`const double y = dx - X_pred[x];` 

Unfortunately this is a fast changing value: 

dx is like that at 44100 Hz  
0.0
0.0
0.0 
0.0 
0.00025
0.0
0.0
0.0
0.0
0.0
0.0

X_pred[x] is like 0,000025 (or -) I am still struggling here ...

At 96 kHz we have still 0.00025 but double as much zeros X_pred[x] has only halve the value.  
This means that we still a samplerate dependency here, because in most cases "y = -X_pred[x]"

```suggestion
                6e-4,   /* medium threshold  */
```

Reply:
The filter knows about the samplerate as $dt = \frac{1}{samplerate}$. Everything works well. Please try this out yourself before making claims like this!